src/main/java/org/dependencytrack/tasks/BomUploadProcessingTask.java

```diff
@@ -109,8 +109,11 @@ public void inform(final Event e) {
                         if (project.getClassifier() == null) {
                             final var classifier = Optional.ofNullable(cycloneDxBom.getMetadata())
                                 .map(org.cyclonedx.model.Metadata::getComponent)
+                                .map(org.cyclonedx.model.Metadata::getSupplier) /**Issue #2373, #2737 */
+                                .map(org.cyclonedx.model.Metadata::getManufacture)
                                 .map(org.cyclonedx.model.Component::getType)
                                 .map(org.cyclonedx.model.Component.Type::name)
+                                .map(org.cyclonedx.model.Component::getSupplier) /**Issue #2373, #2737 */
                                 .map(Classifier::valueOf)
                                 .orElse(Classifier.APPLICATION);
```

Comment:
This entire statement is only intended to extract the component's type, and map it to DT's data model (`Classifier`).

You can think of the `Optional.ofNullable(...).map(...).map(...)` statement like this, but null-safe:

```java
String cdxComponentType = cycloneDxBom.getMetadata().getComponent().getType().name();

Classifier dtClassifier;
if (cdxComponentType == null) {
    dtClassifier = Classifier.APPLICATION; // Fallback when the component has no type
} else {
    dtClassifier = Classifier.valueOf(cdxComponentType);
}

project.setClassifier(dtClassifier);
```

Thus, you should remove the `getSupplier` and `getManufacture` here, as they'll cause compilation errors.

Instead, you'll need a similar logic to what you wrote in `ModelConverter`, but instead of setting the resulting `OrganizationalEntity` on a `Component`, you'll have to set it on `project`. 

It could make sense to move the `OrganizationalEntity` mapping logic into a dedicated method in `ModelConverter`; That way you can reuse it.

Reply:
removed code.  and created manufacture logic in the ModelConverter.  In the interest of time, and lack of understanding ;), ive not made it into a single method yet.